cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -1066,16 +1098,36 @@ std::vector<std::pair<VersionedItem, FrameAndDescriptor>> LocalVersionedEngine::
         return temp_batch_read_internal_direct(stream_ids, version_queries, read_queries, read_options);
     }
 
-    std::vector<folly::Future<std::pair<VersionedItem, FrameAndDescriptor>>> results_fut;
+    std::vector<std::variant<ReadVersionOutput, DataError>> read_versions_or_errors;
+    read_versions_or_errors.reserve(stream_ids.size());
     for (size_t idx=0; idx < stream_ids.size(); idx++) {
         auto version_query = version_queries.size() > idx ? version_queries[idx] : VersionQuery{};
         auto read_query = read_queries.size() > idx ? read_queries[idx] : ReadQuery{};
-        results_fut.push_back(read_dataframe_version_internal(stream_ids[idx],
-                                                              version_query,
-                                                              read_query,
-                                                              read_options));
+        // TODO: https://github.com/man-group/ArcticDB/issues/241
+        // Remove this try-catch in favour of the implementation in temp_batch_read_internal_direct as part of #241
```

Comment:
241 will remove temp_batch_read_internal_direct surely?

Reply:
Yes, but the resulting code will have a `folly::collectAll` in it like `temp_batch_read_internal_direct`, rather than this messy loop-try-catch